packages/plugin-compiler/src/main/kotlin/io/realm/kotlin/compiler/AccessorModifierIrGeneration.kt

```diff
@@ -676,18 +725,27 @@ class AccessorModifierIrGeneration(private val pluginContext: IrPluginContext) {
                     "String" -> PropertyType.RLM_PROPERTY_TYPE_STRING
                     "RealmInstant" -> PropertyType.RLM_PROPERTY_TYPE_TIMESTAMP
                     "ObjectId" -> PropertyType.RLM_PROPERTY_TYPE_OBJECT_ID
-                    else ->
-                        if (inheritsFromRealmObject(type.supertypes())) {
+                    else -> {
```

Comment:
Not sure I understand why this has to change? I assume it is because of the change in `inheritsFromRealmObject`, but I don't understand why that was changed?

Reply:
It's because sets don't support embedded objects. Previously we would consider a property was an object if it was either an ordinary or an embedded object. I needed to make the distinction between the two now so that we could reject sets of embedded objects.

Reply:
But this looks like you did exactly the same logic as before, just in a different way!? Maybe the naming of the original method should just have been updated to `inheritsFromBaseRealmObject` 🤔  

But maybe a test for compilation errors on sets of embedded objects would reveal if the logic is correct ;)

Reply:
Maybe add the `inheritsFromRealmObject` as a utility method on top of the other more explicit ones to keep the logic centralized.

Reply:
I will unify both helpers in one:
```
    private fun inheritsFromRealmObject(
        supertypes: Collection<KotlinType>,
        considerEmbedded: Boolean = false
    ): Boolean = supertypes.any {
        val objectFqNames = when {
            considerEmbedded -> io.realm.kotlin.compiler.realmEmbeddedObjectInterface
            else -> io.realm.kotlin.compiler.realmObjectInterface
        }
        it.constructor.declarationDescriptor?.fqNameSafe in objectFqNames
    }
```
and revert the original function calls since it seems confusing. Only new logic requiring knowing whether a class is embedded or not will use the new codepath.